src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs

```diff
@@ -11,11 +12,13 @@ namespace System.Reflection
     public sealed class ConstructorInvoker
     {
         private readonly MethodBaseInvoker _methodBaseInvoker;
+        private readonly int _parameterCount;
```

Comment:
Do we need a test under System.Reflection for this?

Reply:
Yes thanks. Also made validation consistent across Native and Core.